### PR TITLE
chore(supabase_flutter): Updates README.md to include instructions on how to add native Google login

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

Also includes some clean up of the README.md file. Preview of the new README.md file [here](https://github.com/supabase/supabase-flutter/blob/chore/readme/packages/supabase_flutter/README.md).

Related https://github.com/supabase/supabase-flutter/issues/399

### CODE REVIEW

**Comment by dshukertjr** [comment]
We should be good to merge this one. If someone could review it, it would be amazing! Once this one is merged, we should be able to merge https://github.com/supabase/supabase-flutter/pull/522 as well.

**Comment by DanMossa** [comment]
Does it make sense for us to have platform specific deep link help that's not Supabase specific?

I ask because it seems like maintaining this whenever Apple or Google change something sounds like a pain.

You've already linked a guide.

Just curious!

**Comment by dshukertjr** [comment]
@DanMossa 
>Does it make sense for us to have platform specific deep link help that's not Supabase specific?

Do you mean to have a "Setup Google login on Flutter" and "Setup Apple login on Flutter" type of guide on `supabase.com/docs` somewhere?

**Comment by DanMossa** [comment]
Oops I misread the readme.

**Comment by DanMossa** [comment]
@dshukertjr 

An example of what I mean is if you look at the For Windows section, you have a link to a guide on how to set it up, but then there's also copy/pasted instructions below that.

I'm saying what's the benefit of also copy pasting instructions if the link has the instructions already. It just seems like 1 other thing to maintain when we don't have to.

Lol sorry if I'm not being super clear

File: packages/supabase_flutter/README.md [hunk 1]
```diff
@@ -41,124 +41,213 @@ void main() async {
...
+android {
+    ...
+    defaultConfig {
+        ...
+        manifestPlaceholders += [
+                'appAuthRedirectScheme': applicationId
+        ]
+    }
+}
+```
```
Comment (dshukertjr) [inline packages/supabase_flutter/README.md:113]: @DanMossa So basically we probably should get rid of this section, right? Are there anything else in this PR that feel like is repeating the descriptions on the underlining dependencies?

**Comment by dshukertjr** [comment]
@DanMossa Done!

File: packages/supabase_flutter/README.md [hunk 2]
```diff
@@ -252,86 +326,238 @@ myChannel.on(
...
-  Widget build(BuildContext context) {
-    return FutureBuilder(
-      future: _future,
-      builder: (context, snapshot) {
-        // return your widget with the data from snapshot
-      },
-    );
-  }
-}
+You need to setup deep links when want any of the followings
```
Comment (Vinzent03) [inline packages/supabase_flutter/README.md:355]:
```suggestion
```
Can't this be removed, because it's the same as the heading?

**Review by Vinzent03 — CHANGES_REQUESTED** [review]
I wouldn't remove the `signInWithApple` doc here, but aside from that, LGTM. Much cleaner now!

File: packages/supabase_flutter/README.md (hunk 2 above)
Comment (dshukertjr) [reply packages/supabase_flutter/README.md:355]: Sure, sounds good!

File: packages/supabase_flutter/README.md [hunk 3]
```diff
@@ -41,108 +41,186 @@ void main() async {
...
+  // Do something when there is an auth event
+});
+```
 
-  @override
-  void dispose() {
-    _authSubscription.cancel();
-    super.dispose();
-  }
+#### Native Apple Sign in
```
Comment (dshukertjr) [inline packages/supabase_flutter/README.md:74]: @Vinzent03 
The `signInWithApple()` instructions were not removed in this PR, but moved here!

Comment (Vinzent03) [reply packages/supabase_flutter/README.md:74]: Oh sorry must have overseen this.

**Review by Vinzent03 — APPROVED** [review]

**Pull request merged**